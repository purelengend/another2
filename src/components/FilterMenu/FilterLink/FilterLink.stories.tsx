import { Meta, Story } from '@storybook/react'
import FilterLink from './'
import phoneIcon from '@assets/icons/Phone_Icon.svg'
import laptopIcon from '@assets/icons/Laptop_Icon.svg'
import refrigeratorIcon from '@assets/icons/Refrigerator_Icon.svg'
import tvIcon from '@assets/icons/TV_Icon.svg'
import airConditionerIcon from '@assets/icons/Air_Conditioner_Icon.svg'

export default {
  title: 'Practice Two/FilterLink Component',
  component: FilterLink,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
} as Meta

const Template: Story = args => (
  <FilterLink href={''} icon={''} label={''} {...args} />
)

export const PhoneFilter = Template.bind({})
PhoneFilter.args = {
  href: '#',
  icon: phoneIcon,
  label: 'Phone',
}

export const LaptopFilter = Template.bind({})
LaptopFilter.args = {
  href: '#',
  icon: laptopIcon,
  label: 'Laptop',
}

export const RefrigeratorFilter = Template.bind({})
RefrigeratorFilter.args = {
  href: '#',
  icon: refrigeratorIcon,
  label: 'Refrigerator',
}

export const TVFilter = Template.bind({})
TVFilter.args = {
  href: '#',
  icon: tvIcon,
  label: 'TV',
}

export const AirConditionerFilter = Template.bind({})
AirConditionerFilter.args = {
  href: '#',
  icon: airConditionerIcon,
  label: 'Air Conditioner',
}
