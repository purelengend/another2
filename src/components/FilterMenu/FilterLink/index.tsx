import { memo } from 'react'
import { Link, Text, Image } from '@chakra-ui/react'

interface FilterLinkProps {
  href: string
  icon: string
  label: string
}

const FilterLink = ({ href, icon, label }: FilterLinkProps) => (
  <Link
    href={href}
    className="filter-item"
    fontFamily="Oswald-Regular"
    opacity="0.5"
    _hover={{ textDecoration: 'unset', opacity: '1' }}
  >
    <Image
      src={icon}
      w="68px"
      alt={`The ${label} icon`}
      margin="0 auto"
      loading="eager"
    />
    <Text as="span" display="flex" justifyContent="center">
      {label}
    </Text>
  </Link>
)

export default memo(FilterLink)
