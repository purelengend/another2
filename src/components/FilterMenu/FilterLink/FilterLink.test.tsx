import { render, screen, fireEvent } from '@testing-library/react'
import FilterLink from '.'

describe('FilterLink Component', () => {
  const testProps = {
    href: '/test',
    icon: '/test-icon.png',
    label: 'Test Label',
  }

  it('renders the link with correct href', () => {
    render(<FilterLink {...testProps} />)
    const linkElement = screen.getByRole('link')
    expect(linkElement).toHaveAttribute('href', '/test')
  })

  it('renders the icon with correct alt text', () => {
    render(<FilterLink {...testProps} />)
    const iconElement = screen.getByRole('img')
    expect(iconElement).toHaveAttribute('alt', 'The Test Label icon')
  })

  it('renders the label text', () => {
    render(<FilterLink {...testProps} />)
    const labelElement = screen.getByText('Test Label')
    expect(labelElement).toBeInTheDocument()
  })

  it('applies hover styles on mouse hover', () => {
    render(<FilterLink {...testProps} />)
    const linkElement = screen.getByRole('link')
    fireEvent.mouseEnter(linkElement)
    expect(linkElement).toHaveStyle('text-decoration: unset; opacity: 1')
  })

  it('matches snapshot', () => {
    const { container } = render(<FilterLink {...testProps} />)
    expect(container).toMatchSnapshot()
  })
})
