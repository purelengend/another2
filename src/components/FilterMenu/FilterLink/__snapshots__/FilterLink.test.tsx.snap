// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterLink Component matches snapshot 1`] = `
<div>
  <a
    class="chakra-link filter-item css-zkti8d"
    href="/test"
  >
    <img
      alt="The Test Label icon"
      class="chakra-image css-1q11q9i"
      loading="eager"
      src="/test-icon.png"
    />
    <span
      class="chakra-text css-1l4w6pd"
    >
      Test Label
    </span>
  </a>
</div>
`;
